dreadnode/main.py

```
@@ -163,6 +179,121 @@ def _get_profile_api_key(self, profile: str | None = None) -> str | None:
+            raise RuntimeError("API client is not initialized.")
+
+        try:
+            return self._api.create_workspace(name=name, organization_id=self._organization.id)
+        except RuntimeError as e:
+            if "403: Forbidden" in str(e):
+                raise RuntimeError(
+                    "You do not have permission to create the specified workspace for this organization. (You must be a Contributor to create new workspaces)."
+                ) from e
+            raise
+
+    def _resolve_workspace(self) -> None:
+        if self._api is None:
+            return
+
+        found_workspace: Workspace | None = None
+        if self.workspace:
+            try:
+                found_workspace = self._api.get_workspace(self.workspace)
+            except RuntimeError as e:
+                if "404: Workspace not found" in str(e):
+                    pass  # do nothing, we'll create it below
+                else:
+                    raise
+
+            if not found_workspace and isinstance(self.workspace, UUID):
+                raise RuntimeError(f"Workspace with ID '{self.workspace}' not found.")
+
+            if not found_workspace and isinstance(self.workspace, str):  # specified by name/slug
+                # create the workspace (must be an org contributor)
+                found_workspace = self._create_workspace(name=self.workspace)
+
+        else:  # the user provided no workspace, attempt to find a default one
+            workspaces = self._api.list_workspaces(
+                filters=WorkspaceFilter(
+                    org_id=self._organization.id if self._organization else None
+                )
+            )
+            default_workspace = next((ws for ws in workspaces if ws.is_default is True), None)
+            if default_workspace:
+                found_workspace = default_workspace
+            else:
+                found_workspace = self._create_workspace(name=DEFAULT_WORKSPACE_NAME)
```

consider not auto-create if no default workspace found. The API handles this on org join by invitation or first user.

---

maybe raise an error:

```python
else:
        raise RuntimeError(
            "No default workspace found. This shouldn't happen - please contact org owner."
        )